JPS_VIRTUALSENSOR/AermodAgent/src/main/java/com/cmclinnovations/aermod/QueryClient.java

```diff
@@ -970,22 +974,52 @@ private List<String> getCornersForCitiesQuery(Polygon boundingBox) {
         return corners;
     }
 
+    DSLContext getContext(Connection conn) {
+        return DSL.using(conn, SQLDialect.POSTGRES);
+    }
+
+    boolean tableExists() {
+        String condition = String.format("table_name = '%s'", EnvConfig.ELEVATION_TABLE);
+        boolean tableCheck = false;
+        try (Connection conn = rdbStoreClient.getConnection()) {
+            tableCheck = getContext(conn).select(DSL.count()).from("information_schema.tables").where(condition)
+                    .fetchOne(0,
+                            int.class) == 1;
+        } catch (SQLException e) {
+            LOGGER.error(e.getMessage());
+            return false;
+        }
+
+        return tableCheck;
+    }
+
     public void setElevation(List<StaticPointSource> pointSources, List<Building> buildings, int simulationSrid) {
 
+        String elevationTable = EnvConfig.ELEVATION_TABLE;
+
         for (int i = 0; i < pointSources.size(); i++) {
             StaticPointSource ps = pointSources.get(i);
             String originalSrid = "EPSG:" + ps.getLocation().getSRID();
             double[] xyOriginal = { ps.getLocation().getX(), ps.getLocation().getY() };
             double[] xyTransformed = CRSTransformer.transform(originalSrid, "EPSG:" + simulationSrid, xyOriginal);
             String sqlString = String.format("SELECT ST_Value(rast, ST_SetSRID(ST_MakePoint(%f,%f),%d)) AS val " +
-                    "FROM elevation " + "WHERE ST_Intersects(rast, ST_SetSRID(ST_MakePoint(%f,%f),%d));",
-                    xyTransformed[0], xyTransformed[1], simulationSrid, xyTransformed[0], xyTransformed[1],
+                    "FROM %s WHERE ST_Intersects(rast, ST_SetSRID(ST_MakePoint(%f,%f),%d));",
+                    xyTransformed[0], xyTransformed[1], simulationSrid, elevationTable, xyTransformed[0],
+                    xyTransformed[1],
                     simulationSrid);
 
-            try (Statement stmt = rdbStoreClient.getConnection().createStatement()) {
-                ResultSet result = stmt.executeQuery(sqlString);
-                double elevation = result.getDouble("val");
-                ps.setElevation(elevation);
+            try (Connection conn = rdbStoreClient.getConnection();
+                    Statement stmt = conn.createStatement();
+                    ResultSet result = stmt.executeQuery(sqlString)) {
+                if (result.next()) {
+                    double elevation = result.getDouble("val");
+                    ps.setElevation(elevation);
+                } else {
+                    LOGGER.warn(
+                            "Could not find elevation data for the point source located at ({},{}) in EPSG:{} coordinates. Its elevation will be set to zero when running BPIPPRM and AERMOD.",
+                            xyTransformed[0], xyTransformed[1], simulationSrid);
+                }
```

Comment:
You should not open and close connections in a loop, instead collect all the coordinates you need first, then query in 1 go

Reply:
alternatively, you can open the connection first before you start the loops, i.e. put the loop within the try block, this might require less changes on the current code, ask if unclear

Reply:
I implemented the second suggestion in commit e20a668.